integration/aggregate_compat_test.go

```diff
@@ -100,7 +103,15 @@ func testAggregateStagesCompat(t *testing.T, testCases map[string]aggregateStage
 					require.NoError(t, targetCursor.All(ctx, &targetRes))
 					require.NoError(t, compatCursor.All(ctx, &compatRes))
 
-					AssertEqualDocumentsSlice(t, compatRes, targetRes)
+					if ignoreOrder {
+						// 1. order of result in compat and target are different because
+						// aggregation can assign BSON array to _id.
+						// 2. compat and target may return different number type
+						// because aggregation equates numbers regardless of the type.
```

Comment:
So should we, now or later, but then let's create a separate test, not complicate that test even more with more logic and more helpers

Reply:
removed helpers and created issue :hugs: